### DISCUSSION TITLE
Write up of fMRI prep method & preprocessing rsfMRI without Freesurfer

### DISCUSSION THREAD

**Discussion by jb21469** [HALFpipe/HALFpipe#711] [Q&A] [answered]
Hello,

I am processing a set of resting state data through ENGIMA HALFpipe (v1.2.2) via singularity. I have a few questions regarding preprocessing:

1. Is it correct that you need to place a copy of the freesurfer license within your working director for freesurfer to be able to run during preprocessing? If Freesurfer is not used, how would this affect preprocessing/denoising? 
2. When using fMRIprep, there is typically a write up on the HTML report that describes the preprocessing steps applied. Is there a similar write up for halfpipe describing fMRIprep preprocessing through halfpipe denoising? If not, where can we get information about which fMRIprep steps were run via halfpipe?

Thank you,
Jenna

**Comment by nadza-dz** [comment, marked as answer]
Hello Jenna,

1. Yes, it is correct that you need to place a copy of a FreeSurfer license in the working directory - without it HALFpipe will give an error early on in processing that the license is not found and will stop running, so it is necessary to have this file.
2. There is currently no write-up like that of fMRIprep for HALFpipe, but it is something we will consider for the future! At the moment, you can refer to the [HALFpipe paper](https://onlinelibrary.wiley.com/doi/10.1002/hbm.25829) Figure 2 for all the steps that are applied in a typicial pipeline and adjust this according to your needs. For an example of how this was done in a recent resting-state paper you can have a look at [Bruin et al., 2023](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10827654/).

Hope this helps!
Nadza

**Reply by jb21469** [reply]
Hi Nadza,

Thanks for your reply! That is helpful. I had not placed a copy of the freesurfer license in my working directory but the pipeline ran all the way through. All subjects in reportpreproc.txt say done and correlation matrices were output. Should the pipeline not have proceeded?

For denoising, I selected ICA-AROMA, WM, and CSF. Would this be affected by not having freesurfer in the preprocessing?

Thanks,
Jenna

**Reply by nadza-dz** [reply]
Hi Jenna,

It seems that sometimes the pipeline does indeed run even if the Freesurfer license is not found. I am not certain yet why the warning does not appear in this case, but it does not affect the preprocessing (or denoising) in any way, so you can proceed with the correlation matrices you have obtained.

Best,
Nadza
